cmd/machine-api-operator/controller_context.go

```diff
@@ -29,14 +31,17 @@ type ControllerContext struct {
 func CreateControllerContext(cb *ClientBuilder, stop <-chan struct{}, targetNamespace string) *ControllerContext {
 	kubeClient := cb.KubeClientOrDie("kube-shared-informer")
 	configClient := cb.OpenshiftClientOrDie("config-shared-informer")
+	machineClient := cb.MachineClientOrDie("machine-shared-informer")
 
 	kubeNamespacedSharedInformer := informers.NewSharedInformerFactoryWithOptions(kubeClient, resyncPeriod()(), informers.WithNamespace(targetNamespace))
 	configSharedInformer := configinformersv1.NewSharedInformerFactoryWithOptions(configClient, resyncPeriod()())
+	machineSharedInformer := machineinformersv1beta1.NewSharedInformerFactoryWithOptions(machineClient, resyncPeriod()())
```

Comment:
can we please make this namespaced? also where is this informer waiting to be synced e.g `cache.WaitForCacheSync`, `*.Informer().HasSynced`